### PR TITLE
Sort indices in face_nodes of triangular grids

### PR DESCRIPTION
**Pull request by wmboon**
## Proposed changes

#1507 formed a breaking change for us at PyGeoN. If you perform `StructuredTriangleGrid.compute_geometry`, then we see that it is not oriented properly. 
This fix imposes the assumption that the `face_nodes` is oriented from low to high node index. This allows us to quickly calculate the cell volumes through a matrix-vector product. If this causes negative cell volumes, then we switch back to the implementation based on convex cells

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [x] Minor change (e.g., dependency bumps, broken links).
- [x] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [ ] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Comment by keileg** [comment]
> @keileg All set!

Great. It passes my extra test as well, so maybe we have converged. I'll do a review.

File: src/porepy/grids/grid.py [hunk 1]
```diff
@@ -491,24 +484,49 @@ def _compute_geometry_2d(self) -> None:
...
+            # In an edge case where exactly half the domain is oriented oppositely,
+            # the computed plane normal will be zero. We consider that "unoriented" and
+            # move on.
+            if len_normal < 1e-5 * np.mean(self.face_areas) ** 2:
+                is_oriented = False
+            else:
+                # Normalize the vector
+                plane_normal /= len_normal
+
+        if not is_oriented:
```
Comment (keileg) [inline src/porepy/grids/grid.py]: This can be an 'else', right? If so, I think that is more expressive.

File: src/porepy/grids/grid.py [hunk 2]
```diff
@@ -491,24 +484,49 @@ def _compute_geometry_2d(self) -> None:
...
+            # cell_faces and face_nodes are consistently oriented; in practice, this
+            # means that nodes that are oriented counter clock-wise give positive values
+            subsimplex_volumes = np.dot(plane_normal, subsimplex_normals)
+
+            # In case of an oriented grid, we can quickly compute the cell volumes
+            self.cell_volumes = np.bincount(cellno, weights=subsimplex_volumes)
+
+            # If the orientation flips between subdomains, then negative cell volumes
+            # occur. We consider that "unoriented" and move on.
+            is_oriented = np.all(self.cell_volumes >= 0)
```
[the remark this answers is not in the document]
Comment (wmboon) [reply src/porepy/grids/grid.py]: The first check only verifies that each cell forms a closed node loop, locally. This was sufficient until last week, when subdomains with different orientations started appearing. This PR introduces two additional checks to ensure global orientation consistency.

Annoyingly, a flip in orientation between disconnected subdomains can only be conclusively proven once the cell volumes have been computed. If that is the case, we fall back on the legacy implementation that flips normals.
